src/d/actor/d_a_player_battle.inc

```diff
@@ -1,98 +1,518 @@
 /**
  * d_a_player_battle.inc
  *
- * Code relating to (?) (How is this different from the "cut" functions in the sword file?)
- * 
- * This file is not a standalone translation unit and is instead directly 
+ * Code relating to (?) (How is this different from the "cut" functions in the
+ * sword file?)
+ *
+ * This file is not a standalone translation unit and is instead directly
  * included into d_a_player_main.cpp.
- * 
- * The original name of this file is not known, but a best guess was taken 
+ *
+ * The original name of this file is not known, but a best guess was taken
  * based on the original names of the functions it contains.
  */
 
+#include "d/actor/d_a_player_HIO.h"
 #include "d/actor/d_a_player_main.h"
 
 /* 8013E250-8013E384       .text setSpecialBattle__9daPy_lk_cFi */
-void daPy_lk_c::setSpecialBattle(int) {
-    /* Nonmatching */
+void daPy_lk_c::setSpecialBattle(int param_0) {
+    /* Nonmatching - regalloc */
+    f32 fVar1;
+    fopEn_enemy_c *pfVar3;
+    f32 dVar3;
+    f32 dVar4;
+
+    if ((param_0 == 0) || (!strcmp(dComIfGp_getStartStageName(), "GTower"))) {
+        if (mpAttnEntryA != NULL) {
+            if (mpAttnEntryA->mType == 2) {
+                if (pfVar3 = (fopEn_enemy_c *)mpAttnActorLockOn,
+                    pfVar3 != NULL) {
+                    if (pfVar3->group == 2) {
+                        if (!checkPlayerGuard()) {
+                            if (daPy_dmEcallBack_c::m_type != 1) {
+                                if (mEquipItem == daPyItem_SWORD_e) {
+                                    if (pfVar3->getBtAttackType() != 0 &&
+                                        (pfVar3->getBtStartFrame() -
+                                             daPy_HIO_battle_c0::m.field_0x4 <=
+                                         pfVar3->getBtNowFrame())) {
+                                        if (fVar1 < pfVar3->getBtEndFrame()) {
+                                            dVar4 = pfVar3->getBtMaxDis();
+                                            dVar3 =
+                                                fopAcM_searchActorDistanceXZ(
+                                                    this, pfVar3);
+                                            if (dVar4 > dVar3) {
```

Comment:
Are these temps needed to get the load order right or does something like this match?
```suggestion
                                            if (fopAcM_searchActorDistanceXZ(this, pfVar3) < pfVar3->getBtMaxDis()) {
```